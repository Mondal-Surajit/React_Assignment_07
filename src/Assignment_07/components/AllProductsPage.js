import React from 'react'
import ProductsList from './ProductsList'
import ProductApi from '../data/ProductApi';

export default class AllProductsPage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            products : []
        }
    }

    componentDidMount() {
        this.setState({products: ProductApi.getAllproduct()});
    }

    render() { 
        return(
            <>
                <h2>Products List</h2>
                <ProductsList products={this.state.products} />
            </>
        );

    }
}