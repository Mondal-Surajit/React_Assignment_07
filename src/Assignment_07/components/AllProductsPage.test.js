import React from 'react'
import {mount, shallow} from 'enzyme'
import AllProductsPage from './AllProductsPage'
import toJson from 'enzyme-to-json';


describe('AllProductsPage Snapshot', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = mount(<AllProductsPage/>);
    });

    test('renders correctly', () => {
        expect(toJson(wrapper)).toMatchSnapshot(); 
    }); 
     
});

describe('AllProductsPage rendering of elements' , () => {
    let wrapper1;
    beforeEach(() => { 
        wrapper1 = shallow(<AllProductsPage/>)
    })
    it('renders correct heading for Products List', () => {
        expect(wrapper1.find('h2').render().text()).toEqual("Products List");

    })  
    it('renders one ProductList React component', () => {
        expect(wrapper1.find('ProductsList').length).toEqual(1)
    })  
});

 