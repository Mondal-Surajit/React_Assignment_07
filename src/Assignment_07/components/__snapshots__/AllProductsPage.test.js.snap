// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllProductsPage Snapshot renders correctly 1`] = `
<AllProductsPage>
  <h2>
    Products List
  </h2>
  <ProductsList
    products={
      Array [
        Object {
          "id": 1,
          "name": "Moto G5",
          "price": "Rs. 13000",
          "quantity": "2",
        },
        Object {
          "id": 2,
          "name": "Racold Geyser",
          "price": "Rs. 6000",
          "quantity": "3",
        },
        Object {
          "id": 3,
          "name": "Dell Inspiron",
          "price": "Rs. 50000",
          "quantity": "4",
        },
      ]
    }
  >
    <table>
      <thead>
        <tr>
          <th>
            ID
          </th>
          <th>
            Product Name
          </th>
          <th>
            Quantity
          </th>
          <th>
            Price
          </th>
        </tr>
      </thead>
      <tbody>
        <Product
          id={1}
          key="1"
          name="Moto G5"
          quantity="2"
        >
          <tr>
            <td>
              1
            </td>
            <td>
              Moto G5
            </td>
            <td>
              2
            </td>
            <td>
              Rs. 13000
            </td>
          </tr>
        </Product>
        <Product
          id={2}
          key="2"
          name="Racold Geyser"
          quantity="3"
        >
          <tr>
            <td>
              2
            </td>
            <td>
              Racold Geyser
            </td>
            <td>
              3
            </td>
            <td>
              Rs. 6000
            </td>
          </tr>
        </Product>
        <Product
          id={3}
          key="3"
          name="Dell Inspiron"
          quantity="4"
        >
          <tr>
            <td>
              3
            </td>
            <td>
              Dell Inspiron
            </td>
            <td>
              4
            </td>
            <td>
              Rs. 50000
            </td>
          </tr>
        </Product>
      </tbody>
    </table>
  </ProductsList>
</AllProductsPage>
`;
